Refactors stripcode, the oldest, crappiest, and most bizarre code I have ever seen

9600bauds opened the pull request:
And does a lot of housekeeping around inventorycode to make it less of a pain in the ass to eventually refactor. I swear this shit was worse than lifecode.
## Changelog

Trying to put an item in the wrong slot via the stripping menu now gives an explicit error message, so as to not waste your time.
Buffed the Pickpocket's Gloves: Their effects now apply to all storage slots, not just ID and pockets.
Fixed items in your pocket slots not applying their on_found() effects (e.g. mousetraps) when stripped individually, only when your jumpsuit was stripped.
Fixed items in your pocket slots not applying their on_stripped() effects (e.g. SPS) when your jumpsuit was stripped, only individually.
You can now toggle someone's suit sensors via the stripping menu, without having to remove their jumpsuit.
There is now one unified option to toggle other people's internals via the stripping menu, which is available even if they have a tank in their pockets or hands.
Simplified the way toggling internals chooses a tank: It will now simply look in your suit storage, then back, then belt, then pockets, then hands. It will also only take internals from a jetpack as last option.
Removes the ability to perform CPR on alien larvas. Sorry guys.
Accidentally fixed bug where you could spam CPR and do CPR multiple times at the same time in the same mob. Sorry guys.
## To do
- [x] The changelog
- [ ] See if there's a way to make the inventory screen not reopen itself when refreshing if it was closed
- [x] Test this shit
- [x] Fix monkey HUD not updating when equipping to slot
- [x] Let you know if internals are on or off before setting them via stripping menu
- [x] Fix reversestripping borg modules and cantdrop items
- [x] Find the issues that this closes

Resolves #8222, Fixes #7760, Resolves #9969
- [ ] Remove `src.`
## To test
- [x] Test stripping out of every slot imaginable
- [x] Test stripping out of every slot imaginable, with monkeys
- [x] Test stripping out of every slot imaginable, including naughty href stuff
- [x] Test stripping into every slot imaginable
- [x] Test stripping into every slot imaginable, with monkeys
- [x] Test stripping into every slot imaginable, including naughty href stuff
- [x] DNA Injector: With monkeys, test logging with block and general injectors to see if it picks up monkeying
- [x] Cuffs: As borg, test logging, test resisting out of
- [x] Cuffs: Test breaking with hulk and freedom implant
- [x] Syndicuffs, test if they play nice when detonating and when explodified etc
- [x] Reverse stripping borg modules, including emag modules
- [x] Stripping undroppable items, including reverse stripping
- [x] Reverse stripping 2handed items in improper circumstances, also stripping offhand items
- [x] Internals, internals logging
- [x] Setting suit sensors
- [x] Canister logging
- [x] Pocket on_found and stripped behavior, when dropped via uniform stripping and when stripped directly
- [x] Test Diona and other monkey child's UI, stripping for shit they can't wear
- [x] Test naughty href stuff
- [x] CPR
## Concerning comments I found while working on this PR

![atom_2016-06-03_19-57-20](https://cloud.githubusercontent.com/assets/6526157/15967472/456f37c2-2efe-11e6-8e75-555c2824fcbb.png)
![atom_2016-06-03_20-23-40](https://cloud.githubusercontent.com/assets/6526157/15967475/477b50c8-2efe-11e6-8b92-3c917a2b1c3e.png)
![atom_2016-06-03_20-24-46](https://cloud.githubusercontent.com/assets/6526157/15967478/498aabc0-2efe-11e6-82de-b1d806d217d2.png)
![atom_2016-06-05_16-27-39](https://cloud.githubusercontent.com/assets/6526157/15967481/4eb7c772-2efe-11e6-878b-c61e1a45fc3a.png)
![atom_2016-06-05_16-47-17](https://cloud.githubusercontent.com/assets/6526157/15967485/5109516c-2efe-11e6-9d15-5105791b3a98.png)
![atom_2016-06-05_18-52-05](https://cloud.githubusercontent.com/assets/6526157/15967488/533fa2f6-2efe-11e6-8c38-b4ebce805804.png)
![atom_2016-06-05_21-45-50](https://cloud.githubusercontent.com/assets/6526157/15967491/55867be8-2efe-11e6-9fd0-680f2774bd86.png)
![atom_2016-06-06_01-39-38](https://cloud.githubusercontent.com/assets/6526157/15967499/5d51c3c8-2efe-11e6-8d65-2ac3a811f67c.png)
~new~
![atom_2016-06-12_21-08-21](https://cloud.githubusercontent.com/assets/6526157/16021989/0667426e-318c-11e6-83e8-2ee3cc02cd27.png)

DrCelt commented on code/_onclick/hud/monkey.dm:
```diff
@@ -142,41 +142,33 @@
+	inv_box.icon_state = "center"
+	inv_box.screen_loc = ui_monkey_uniform
+	inv_box.layer = 19
+	src.other += inv_box
```
no need for src.

DrCelt commented on code/_onclick/hud/monkey.dm:
```diff
@@ -142,41 +142,33 @@
+	inv_box.screen_loc = ui_monkey_hat
+	inv_box.slot_id = slot_head
+	inv_box.layer = 19
+	src.other += inv_box
```
no need for src.

9600bauds commented:
stop

DrCelt commented on code/_onclick/hud/monkey.dm:
```diff
@@ -279,7 +271,7 @@
 
-	mymob.client.screen += list( mymob.throw_icon, mymob.zone_sel, mymob.m_hatbg, mymob.m_hat, mymob.m_suitclothesbg, mymob.m_suitclothes, mymob.m_glassesbg, mymob.m_glasses, mymob.oxygen, mymob.pressure, mymob.toxin, mymob.bodytemp, mymob.internals, mymob.fire, mymob.healths, mymob.pullin, mymob.gun_setting_icon) //, mymob.hands, mymob.rest, mymob.sleep, mymob.mach )
+	mymob.client.screen += list( mymob.throw_icon, mymob.zone_sel, mymob.oxygen, mymob.pressure, mymob.toxin, mymob.bodytemp, mymob.internals, mymob.fire, mymob.healths, mymob.pullin, mymob.gun_setting_icon) //, mymob.hands, mymob.rest, mymob.sleep, mymob.mach )
 	mymob.client.screen += src.adding + src.other
```
no need for src.

unid15 replied:
that's one of my most favorite github memes

9600bauds commented:
stop that

DrCelt commented on code/game/machinery/atmoalter/canister.dm:
```diff
@@ -357,28 +355,21 @@
-				if(src.air_contents.toxins > 0 || (istype(S)))
-					message_admins("[usr.real_name] ([formatPlayerPanel(usr,usr.ckey)]) opened a canister that contains \[[src.air_contents.toxins > 0 ? "Toxins" : ""] [istype(S) ? " N2O" : ""]\] at [formatJumpTo(loc)]!")
-					log_admin("[usr]([ckey(usr.key)]) opened a canister that contains \[[src.air_contents.toxins > 0 ? "Toxins" : ""] [istype(S) ? " N2O" : ""]\] at [loc.x], [loc.y], [loc.z]")
+				var/naughty_stuff = air_contents.loggable_contents()
```
muh atomic prs though

PJB3005 commented on code/game/objects/items.dm:
```diff
@@ -634,12 +641,30 @@
+			if(slot_head)
+				if(MO.hat)
+					return 0
+				if( !(slot_flags & SLOT_HEAD) )
```
~~`if(slot_flags ^ SLOT_HEAD)`~~

ComicIronic replied:
Is that functionally equivalent? 00010 ^ 00010 = 00000, but !(00010 & 00010) = 11101

clusterfack replied:
You are thinking of ~ comic

PJB3005 replied:
Never mind me, it isn't, I'm an idiot.

9600bauds replied:
So should I leave this as is?

clusterfack replied:
Yeah its fine, DELETE THIS

PJB3005 commented on code/game/objects/items.dm:
```diff
@@ -634,12 +641,30 @@
+			if(slot_glasses)
+				if(MO.glasses)
+					return 0
+				if( !(slot_flags & SLOT_EYES) )
```
`if(slot_flags ^ SLOT_EYES)`

unid15 commented:
Great job!

PJB3005 commented on code/game/objects/items.dm:
```diff
@@ -634,12 +641,30 @@
+			if(slot_w_uniform)
+				if(MO.uniform)
+					return 0
+				if( !(slot_flags & SLOT_ICLOTHING) )
```
~~`if(slot_flags ^ SLOT_ICLOTHING)`~~

9600bauds replied:
Where can I find the documentation on these fabled bitflags?

Exxion replied:
In the DM ref, along with everything else in DM
But that one's XOR, which is only equivalent because the second one is guaranteed to not be 0

9600bauds replied:
Where in the DM ref
I searched for operators, bitwise operators, couldn't find it
I'm sure it's there, that's why I'm asking where

Exxion replied:
They're in with the regular operators; no distinction is made
Which is stupid, but oh well

DrCelt commented on code/game/objects/items/weapons/handcuffs.dm:
```diff
@@ -48,57 +48,47 @@
-			O.process()
+		var/mob/living/carbon/human/H = C
+		if (!H.has_organ_for_slot(slot_handcuffed))
+			to_chat(user, "<span class='danger'>\The [C] needs at least two wrists before you can cuff them together!</span>")
```
"The Angelite Entyshak needs at least two wrists before you can cuff them together!"
Remove the "the" and note that this is a balance change - assistants like angelite used to take only one hand in order to not get cuffed.

9600bauds replied:
The \the macro doesn't the unless nethessary
Also, the original has_organ_for_slots was down in the ass equip_e system, but rest assured it already worked like that

unid15 replied:
"\The" doesn't appear before proper names, does it? It'll only appear if the mob is named something like "unknown"

DrCelt replied:
balance issue still remains

9600bauds replied:
The check is the same, it was already there, just in another file

DrCelt replied:
It was almost certainly disabled.

9600bauds replied:
Just tested in another branch, it's disabled
No balance change here

DrCelt commented on code/modules/clothing/clothing.dm:
```diff
@@ -436,33 +451,43 @@ BLIND     // can't see anything
-		if(3)
-			to_chat(usr, "<span class='notice'>Your suit will now report your vital lifesigns as well as your coordinate position.</span>")
+	if(is_holder_of(user, src))
+		switch(sensor_mode) //i'm sure there's a more compact way to write this but c'mon
```
there is definitely a more compact way to write this

PJB3005 commented on code/game/objects/items.dm:
```diff
@@ -649,6 +674,42 @@
+					return 0
+				if(AH.l_store)
+					return 0
+				if( !(w_class <= W_CLASS_SMALL || (slot_flags & SLOT_POCKET)) )
```
`if(w_class > W_CLASS_SMALL || slot_flags ^ SLOT_POCKET)`

PJB3005 commented on code/game/objects/items.dm:
```diff
@@ -649,6 +674,42 @@
+					return 0
+				if(AH.r_store)
+					return 0
+				if( !(w_class <= W_CLASS_SMALL || (slot_flags & SLOT_POCKET)) )
```
`if(w_class > W_CLASS_SMALL || slot_flags ^ SLOT_POCKET)`

DrCelt commented on code/modules/mob/living/carbon/alien/larva/larva.dm:
```diff
@@ -191,19 +191,25 @@
 			else
 				if(M.health >= -75.0)
-					if ((M.head && M.head.flags & 4) || (M.wear_mask && !( M.wear_mask.flags & 32 )) )
+					if ((M.head && M.head.flags & 4) || (M.wear_mask && !( M.wear_mask.flags & 32 )) ) //Why can we even give CPR to an alien larva?
```
is this whole thing really copypasted from humancode

9600bauds replied:
Most likely from exceptionally old humancode

PJB3005 commented on code/game/objects/items/weapons/dna_injector.dm:
```diff
@@ -127,90 +126,38 @@
+
+	user.visible_message("<span class='danger'>\The [user] is trying to inject \the [M] with \the [src]!</span>")
+	inuse = 1
+	spawn(50)
```
Remove, reset `inuse` somewhere AFTER the `do_after`

DrCelt commented:
thanks based bauds

PJB3005 commented on code/game/objects/items/weapons/handcuffs.dm:
```diff
@@ -48,57 +48,47 @@
+			cuffs = new(get_turf(user))
+		else
+			user.drop_from_inventory(cuffs)
+		cuffs.loc = C
```
What so there isn't a better way to do this? No helpers?

9600bauds replied:
That's what I get for copypasting from Bay

PJB3005 commented on code/game/objects/items/weapons/handcuffs.dm:
```diff
@@ -48,57 +48,47 @@
 
-	return TRUE
+/obj/item/weapon/handcuffs/proc/handcuffs_remove(var/mob/living/carbon/C)
+	C.handcuffed = null
```
Same deal, helpers.

9600bauds replied:
https://github.com/d3athrow/vgstation13/commit/e1050fd4bbc235b8dbf94d07e20362c6908d8969#diff-c122663326faf1e006f6276e4e9db8adR424

aaaaaaaaaaaa

9600bauds replied:
https://github.com/d3athrow/vgstation13/commit/e1050fd4bbc235b8dbf94d07e20362c6908d8969#diff-59a07870f8bd3a0ebe35d335d49994c6R176

This is why we need this PR

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:
```diff
@@ -210,19 +210,25 @@
 			else
 				if(M.health >= -75.0)
-					if(((M.head && M.head.flags & 4) || ((M.wear_mask && !( M.wear_mask.flags & 32 )) || ((head && head.flags & 4) || (wear_mask && !( wear_mask.flags & 32))))))
+					if(((M.head && M.head.flags & 4) || ((M.wear_mask && !( M.wear_mask.flags & 32 )) || ((head && head.flags & 4) || (wear_mask && !( wear_mask.flags & 32)))))) //Holy fucking christ why
```
I don't know why, but it sounds like the PERFECT time to refactor it!

DrCelt replied:
at the very least change those magic numbers holy shit

9600bauds replied:
Honestly is there any reason why you can give CPR to aliens?
I just copypasted the new CPR code expecting it not to work and was fully prepared to just remove it all when it didn't work
Sadly, it did work

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:
```diff
@@ -210,19 +210,25 @@
+						return 0
+
+					cpr_time = 0
+					spawn(30)
```
Nopenopenope

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:
```diff
@@ -210,19 +210,25 @@
+					M.visible_message("<span class='danger'>\The [M] is trying perform CPR on \the [src]!</span>")
+
+					if(do_after(M, src, 3 SECONDS))
+						adjustOxyLoss(-(min(getOxyLoss(), 7)))
```
`adjustOxyLoss(-min(getOxyLoss(), 7))`

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:
```diff
@@ -325,21 +331,33 @@ In all, this is a lot like the monkey code. /N
 /mob/living/carbon/alien/humanoid/show_inv(mob/user as mob)
-
 	user.set_machine(src)
+	var/pickpocket = usr.isGoodPickpocket()
```
`user`

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/inventory.dm:
```diff
@@ -1,3 +1,44 @@
+	if(!slot) return
+	if(!istype(W)) return
+
+	if(src.is_holding_item(W))
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/inventory.dm:
```diff
@@ -1,3 +1,44 @@
+	if(!istype(W)) return
+
+	if(src.is_holding_item(W))
+		src.u_equip(W)
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/inventory.dm:
```diff
@@ -1,3 +1,44 @@
+
+	switch(slot)
+		if(slot_head)
+			src.head = W
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/inventory.dm:
```diff
@@ -1,3 +1,44 @@
+			src.head = W
+			update_inv_head(redraw_mob)
+		if(slot_wear_suit)
+			src.wear_suit = W
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/inventory.dm:
```diff
@@ -1,3 +1,44 @@
+			src.wear_suit = W
+			update_inv_wear_suit(redraw_mob)
+		if(slot_l_store)
+			src.l_store = W
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/inventory.dm:
```diff
@@ -1,3 +1,44 @@
+			src.l_store = W
+			update_inv_pockets(redraw_mob)
+		if(slot_r_store)
+			src.r_store = W
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/update_icons.dm:
```diff
@@ -92,10 +90,10 @@
 		if(!t_state)	t_state = head.icon_state
-		var/image/lying		= image("icon" = 'icons/mob/mob.dmi', "icon_state" = "[t_state]2")
-		var/image/standing	= image("icon" = 'icons/mob/mob.dmi', "icon_state" = "[t_state]")
+		var/image/lying		= image(((head.icon_override) ? head.icon_override : 'icons/mob/head.dmi'), "icon_state" = "[t_state]")
```
`head.icon_override || 'icons/mob/head.dmi'`

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/update_icons.dm:
```diff
@@ -92,10 +90,10 @@
-		var/image/lying		= image("icon" = 'icons/mob/mob.dmi', "icon_state" = "[t_state]2")
-		var/image/standing	= image("icon" = 'icons/mob/mob.dmi', "icon_state" = "[t_state]")
+		var/image/lying		= image(((head.icon_override) ? head.icon_override : 'icons/mob/head.dmi'), "icon_state" = "[t_state]")
+		var/image/standing	= image(((head.icon_override) ? head.icon_override : 'icons/mob/head.dmi'), "icon_state" = "[t_state]")
```
`head.icon_override || 'icons/mob/head.dmi'`

9600bauds replied:
Please just pretend this whole file doesn't exist, it's not actually used anywhere and I only briefly toyed around with making it work before realizing it's unused for a good reason

ComicIronic replied:
Byond has a null coalesce?

PJB3005 commented on code/modules/mob/living/carbon/alien/larva/larva.dm:
```diff
@@ -191,19 +191,25 @@
+						return 0
+
+					cpr_time = 0
+					spawn(30)
```
Aaaaaah

PJB3005 commented on code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:
```diff
@@ -210,19 +210,25 @@
+
+					if(do_after(M, src, 3 SECONDS))
+						adjustOxyLoss(-(min(getOxyLoss(), 7)))
+						updatehealth()
```
Is this really needed?

PJB3005 commented on code/modules/mob/living/carbon/alien/larva/larva.dm:
```diff
@@ -191,19 +191,25 @@
+
+					if(do_after(M, src, 3 SECONDS))
+						adjustOxyLoss(-(min(getOxyLoss(), 7)))
+						updatehealth()
```
Really needed?

PJB3005 commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -473,9 +426,28 @@
+		if(usr.incapacitated() || !Adjacent(usr)|| isanimal(usr))
+			return
+		set_internals(usr)
+	else ..()
```
Second `..()` call.

PJB3005 commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -473,9 +426,28 @@
+		src << browse(null, t1)
 
+	if(href_list["hands"])
+		if(usr.incapacitated() || !Adjacent(usr)|| isanimal(usr))
```
All these checks should be moved so it's not copy pasted into the other 2

PJB3005 commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -545,145 +435,108 @@
+	user.set_machine(src)
+	var/pickpocket = usr.isGoodPickpocket()
+	var/list/obscured = check_obscured_slots()
+	var/TAB = "&nbsp;&nbsp;&nbsp;&nbsp;"
```
Make this `const`.

PJB3005 replied:
Or a define that works too.

PJB3005 commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -545,145 +435,108 @@
-		var/slot = href_list["item"]
-		var/obj/item/place_item = usr.get_active_hand()
-		var/obj/item/id_item = src.wear_id
+/mob/living/carbon/human/Topic(href, href_list)
```
Is this copy pasted from above? Please no copy paste.

PJB3005 commented on code/modules/mob/living/carbon/human/human_attackhand.dm:
```diff
@@ -233,19 +233,25 @@
+
+			cpr_time = 0
+			spawn(30)
+				cpr_time = 1
```
Oh god.

9600bauds replied:
To be fair, I copypasted that from Bay

PJB3005 commented on code/modules/mob/living/carbon/human/human_attackhand.dm:
```diff
@@ -233,19 +233,25 @@
+
+			if(do_after(M, src, 3 SECONDS))
+				adjustOxyLoss(-(min(getOxyLoss(), 7)))
+				updatehealth()
```
Seriously is this real?

9600bauds commented:
DISCLAIMER: I put src in some places to help with readability, since it helps differentiate the stripp-ee from the stripper
user.name and just "name" blend in too much

PJB3005 commented on code/modules/mob/living/carbon/human/human_attackhand.dm:
```diff
@@ -233,19 +233,25 @@
+			M.visible_message("<span class='danger'>\The [M] is trying perform CPR on \the [src]!</span>")
+
+			if(do_after(M, src, 3 SECONDS))
+				adjustOxyLoss(-(min(getOxyLoss(), 7)))
```
Remove extra parenthesis around `min()`

PJB3005 commented on code/modules/mob/living/carbon/human/inventory.dm:
```diff
@@ -301,14 +301,64 @@
+//unset redraw_mob to prevent the mob from being redrawn at the end.
+/mob/living/carbon/human/equip_to_slot_if_possible(obj/item/W as obj, slot, act_on_fail = 0, disable_warning = 0, redraw_mob = 1, automatic = 0)
+	switch(W.mob_can_equip(src, slot, disable_warning, automatic))
+		if(0)
```
This is gonna need some defines.

PJB3005 commented on code/modules/mob/living/carbon/human/inventory.dm:
```diff
@@ -301,14 +301,64 @@
+					qdel(W)
+					W = null
+				if(EQUIP_FAILACTION_DROP)
+					W.loc=get_turf(src) // I think.
```
`forceMove()`

PJB3005 commented on code/modules/mob/living/carbon/human/inventory.dm:
```diff
@@ -301,14 +301,64 @@
+//set del_on_fail to have it delete W if it fails to equip
+//set disable_warning to disable the 'you are unable to equip that' warning.
+//unset redraw_mob to prevent the mob from being redrawn at the end.
+/mob/living/carbon/human/equip_to_slot_if_possible(obj/item/W as obj, slot, act_on_fail = 0, disable_warning = 0, redraw_mob = 1, automatic = 0)
```
`act_on_fail` needs defines.

9600bauds replied:
It does use defines

PJB3005 commented on code/modules/mob/living/carbon/human/inventory.dm:
```diff
@@ -301,14 +301,64 @@
+					return
+
+				if(drop_item(W))
+					if(!(put_in_active_hand(wearing)))
```
Extra parenthesis

PJB3005 commented on code/modules/mob/living/carbon/internals.dm:
```diff
@@ -0,0 +1,69 @@
+		if(user != src)
+			if(!user.isGoodPickpocket())
+				visible_message("<span class='warning'>\The [user] shuts off \the [src]'s internals!</span>")
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has disabled [src.name]'s ([src.ckey]) internals.</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/internals.dm:
```diff
@@ -0,0 +1,69 @@
+			if(!user.isGoodPickpocket())
+				visible_message("<span class='warning'>\The [user] shuts off \the [src]'s internals!</span>")
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has disabled [src.name]'s ([src.ckey]) internals.</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='red'>Internals disabled by [user.name] ([user.ckey]).</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/internals.dm:
```diff
@@ -0,0 +1,69 @@
+				visible_message("<span class='warning'>\The [user] shuts off \the [src]'s internals!</span>")
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has disabled [src.name]'s ([src.ckey]) internals.</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='red'>Internals disabled by [user.name] ([user.ckey]).</font>")
+			log_attack("[user.name] ([user.ckey]) has disabled [src.name]'s ([src.ckey]) internals.")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/internals.dm:
```diff
@@ -0,0 +1,69 @@
+			var/gas_contents = T.air_contents.english_contents_list()
+			if(!user.isGoodPickpocket())
+				to_chat(user, "<span class='notice'>\The [user] has enabled [src]'s internals.</span>")
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has enabled [src.name]'s ([src.ckey]) internals (Gas contents: [gas_contents]).</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/internals.dm:
```diff
@@ -0,0 +1,69 @@
+			if(!user.isGoodPickpocket())
+				to_chat(user, "<span class='notice'>\The [user] has enabled [src]'s internals.</span>")
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has enabled [src.name]'s ([src.ckey]) internals (Gas contents: [gas_contents]).</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='red'>Internals enabled by [user.name] ([user.ckey]) (Gas contents: [gas_contents]).</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/internals.dm:
```diff
@@ -0,0 +1,69 @@
+				to_chat(user, "<span class='notice'>\The [user] has enabled [src]'s internals.</span>")
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has enabled [src.name]'s ([src.ckey]) internals (Gas contents: [gas_contents]).</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='red'>Internals enabled by [user.name] ([user.ckey]) (Gas contents: [gas_contents]).</font>")
+			log_attack("[user.name] ([user.ckey]) has disabled [src.name]'s ([src.ckey]) internals (Gas contents: [gas_contents]).")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/monkey/inventory.dm:
```diff
@@ -1,199 +1,99 @@
-	if(W == get_active_hand())
-		u_equip(W,0)
+	if(src.is_holding_item(W))
+		src.u_equip(W)
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/monkey/inventory.dm:
```diff
@@ -1,199 +1,99 @@
 
-	if(W == get_active_hand())
-		u_equip(W,0)
+	if(src.is_holding_item(W))
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/monkey/inventory.dm:
```diff
@@ -1,199 +1,99 @@
-			W.equipped(src, slot)
 			update_inv_back(redraw_mob)
+		if(slot_head)
+			src.hat = W
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/monkey/inventory.dm:
```diff
@@ -1,199 +1,99 @@
+			src.hat = W
+			update_inv_hat(redraw_mob)
+		if(slot_w_uniform)
+			src.uniform = W
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/monkey/inventory.dm:
```diff
@@ -1,199 +1,99 @@
+			src.uniform = W
+			update_inv_uniform(redraw_mob)
+		if(slot_glasses)
+			src.glasses = W
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/monkey/monkey.dm:
```diff
@@ -143,45 +143,56 @@
+	var/health_deficiency = (100 - health)
+	if(health_deficiency >= 45) tally += (health_deficiency / 25)
+
+	if (bodytemperature < 283.222)
```
There has to be a define for this.

9600bauds replied:
This is shitcode copypasted from humans, I didn't even touch it, git diff thinks I did because I moved a couple of procs down into the file. Someone was up to a rework of this a while ago, outta scope for me

PJB3005 commented on code/modules/mob/living/carbon/monkey/monkey.dm:
```diff
@@ -143,45 +143,56 @@
+	if(istype(loc,/turf/simulated/floor))
+		var/turf/simulated/floor/T = loc
+
+		if(T.material=="phazon")
```
What the fuck is OOP anyways.

9600bauds replied:
This is shitcode copypasted from humans, I didn't even touch it, git diff thinks I did because I moved a couple of procs down into the file. Someone was up to a rework of this a while ago, outta scope for me

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+	if(do_mob(user, src, HUMAN_STRIP_DELAY)) //Fails if the user moves, changes held item, is incapacitated, etc.
+		if(held.mob_can_equip(src, slot, disable_warning = 1))
+			user.drop_from_inventory(held)
+			src.equip_to_slot(held, slot) //Not using equip_to_slot_if_possible() because we want to check that the guy can wear this before dropping it
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+	var/obj/item/held = user.get_active_hand()
+
+	if(do_mob(user, src, HUMAN_STRIP_DELAY)) //Fails if the user moves, changes held item, is incapacitated, etc.
+		if(src.put_in_hand_check(held, index))
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+	if(do_mob(user, src, HUMAN_STRIP_DELAY)) //Fails if the user moves, changes held item, is incapacitated, etc.
+		if(src.put_in_hand_check(held, index))
+			user.drop_from_inventory(held)
+			src.put_in_hand(index, held)
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			return TRUE
+
+/mob/living/carbon/proc/handle_strip_slot(var/mob/living/user, var/slot)
+	if(slot in src.check_obscured_slots()) //Ideally they wouldn't even get the button to do this, but they could have an outdated menu or something
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		return
+
+	var/obj/item/held = user.get_active_hand()
+	var/obj/item/target_item = src.get_item_by_slot(slot)
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			visible_message("<span class='danger'>\The [user] is trying to remove \a [target_item] from \the [src]'s [src.slotID2slotname(slot)]!</span>")
+
+		if(strip_item_from(user, target_item, slot, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has stripped \a [target_item] from [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			return
+
+		if(!pickpocket)
+			visible_message("<span class='danger'>\The [user] is trying to remove \a [target_item] from \the [src]'s [src.slotID2slotname(slot)]!</span>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+
+		if(strip_item_from(user, target_item, slot, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has stripped \a [target_item] from [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has had \a [target_item] stripped from their [src.slotID2slotname(slot)] by [user.name] ([user.ckey])</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		if(strip_item_from(user, target_item, slot, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has stripped \a [target_item] from [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has had \a [target_item] stripped from their [src.slotID2slotname(slot)] by [user.name] ([user.ckey])</font>")
+			log_attack("[user.name] ([user.ckey]) has stripped \a [target_item] from [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			log_attack("[user.name] ([user.ckey]) has stripped \a [target_item] from [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])")
+			show_inv(user)
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to strip \a [target_item] from [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			show_inv(user)
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to strip \a [target_item] from [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.name] ([user.ckey]) has failed to strip \a [target_item] from this mob's [src.slotID2slotname(slot)]</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to strip \a [target_item] from [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.name] ([user.ckey]) has failed to strip \a [target_item] from this mob's [src.slotID2slotname(slot)]</font>")
+			log_attack("[user.name] ([user.ckey]) has failed to strip \a [target_item] from [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			to_chat(user, "<span class='warning'>You can't put that there!</span>") //Ideally we could have a more descriptive message since this can fail for a variety of reasons, but whatever
+			return
+		if(!src.has_organ_for_slot(slot))
+			to_chat(user, "<span class='warning'>\The [src] has no [src.slotID2slotname(slot)].</span>") //blunt
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		if(!held.mob_can_equip(src, slot, disable_warning = 1)) //This also checks for glued items, target being too fat for the clothing item, and such
+			to_chat(user, "<span class='warning'>You can't put that there!</span>") //Ideally we could have a more descriptive message since this can fail for a variety of reasons, but whatever
+			return
+		if(!src.has_organ_for_slot(slot))
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			visible_message("<span class='danger'>\The [user] is trying to put \a [held] on \the [src]!</span>")
+
+		if(reversestrip_into_slot(user, slot, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has put \a [held] into [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+
+		if(reversestrip_into_slot(user, slot, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has put \a [held] into [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has had \a [held] put into their [src.slotID2slotname(slot)] by [user.name] ([user.ckey])</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		if(reversestrip_into_slot(user, slot, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has put \a [held] into [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has had \a [held] put into their [src.slotID2slotname(slot)] by [user.name] ([user.ckey])</font>")
+			log_attack("[user.name] ([user.ckey]) has put \a [held] into [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			log_attack("[user.name] ([user.ckey]) has put \a [held] into [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])")
+			show_inv(user)
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to place \a [held] into [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			show_inv(user)
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to place \a [held] into [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.name] ([user.ckey]) has failed to place \a [held] into this mob's [src.slotID2slotname(slot)]</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to place \a [held] into [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.name] ([user.ckey]) has failed to place \a [held] into this mob's [src.slotID2slotname(slot)]</font>")
+			log_attack("[user.name] ([user.ckey]) has failed to place \a [held] into [src.name]'s [src.slotID2slotname(slot)] ([src.ckey])")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+	if(!index || !isnum(index))	return
+
+	var/obj/item/held = user.get_active_hand()
+	var/obj/item/target_item = src.held_items[index]
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+	var/obj/item/target_item = src.held_items[index]
+	var/pickpocket = user.isGoodPickpocket()
+
+	to_chat(world, "HITLERS: We are [user], stripping [src], our target item is [target_item] in hand [src.get_index_limb_name(index)] index [index] and our held item is [held] please advice")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			return
+
+		if(!pickpocket)
+			visible_message("<span class='warning'>\The [user] is trying to take \a [target_item] from \the [src]'s [src.get_index_limb_name(index)]!</span>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			visible_message("<span class='warning'>\The [user] is trying to take \a [target_item] from \the [src]'s [src.get_index_limb_name(index)]!</span>")
+
+		if(strip_item_from(user, target_item, null, pickpocket)) //slot is null since it's a hand index
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has stripped \a [target_item] from [src.name] ([src.ckey])'s [src.get_index_limb_name(index)]</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+
+		if(strip_item_from(user, target_item, null, pickpocket)) //slot is null since it's a hand index
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has stripped \a [target_item] from [src.name] ([src.ckey])'s [src.get_index_limb_name(index)]</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has had \a [target_item] stripped from their [src.get_index_limb_name(index)] by [user.name] ([user.ckey])</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		if(strip_item_from(user, target_item, null, pickpocket)) //slot is null since it's a hand index
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has stripped \a [target_item] from [src.name] ([src.ckey])'s [src.get_index_limb_name(index)]</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has had \a [target_item] stripped from their [src.get_index_limb_name(index)] by [user.name] ([user.ckey])</font>")
+			log_attack("[user.name] ([user.ckey]) has stripped \a [target_item] from [user.name]'s ([src.ckey])'s [src.get_index_limb_name(index)]")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			log_attack("[user.name] ([user.ckey]) has stripped \a [target_item] from [user.name]'s ([src.ckey])'s [src.get_index_limb_name(index)]")
+			show_inv(user)
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to strip \a [target_item] from [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			show_inv(user)
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to strip \a [target_item] from [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.name] ([user.ckey]) has failed to strip \a [target_item] from this mob's [src.get_index_limb_name(index)]</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to strip \a [target_item] from [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.name] ([user.ckey]) has failed to strip \a [target_item] from this mob's [src.get_index_limb_name(index)]</font>")
+			log_attack("[user.name] ([user.ckey]) has failed to strip \a [target_item] from [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			log_attack("[user.name] ([user.ckey]) has failed to strip \a [target_item] from [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]")
+
+	else if(istype(held))
+		if(!src.put_in_hand_check(held, index))
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			return
+
+		if(!pickpocket)
+			visible_message("<span class='warning'>\The [user] is trying to put \a [held] on \the [src]'s [src.get_index_limb_name(index)]!</span>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			visible_message("<span class='warning'>\The [user] is trying to put \a [held] on \the [src]'s [src.get_index_limb_name(index)]!</span>")
+
+		if(reversestrip_into_hand(user, index, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has put \a [held] into [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+
+		if(reversestrip_into_hand(user, index, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has put \a [held] into [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has had \a [held] put into their [src.get_index_limb_name(index)] by [user.name] ([user.ckey])</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		if(reversestrip_into_hand(user, index, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has put \a [held] into [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has had \a [held] put into their [src.get_index_limb_name(index)] by [user.name] ([user.ckey])</font>")
+			log_attack("[user.name] ([user.ckey]) has put \a [held] into [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			log_attack("[user.name] ([user.ckey]) has put \a [held] into [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]")
+			show_inv(user)
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to place \a [held] into [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			show_inv(user)
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to place \a [held] into [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.name] ([user.ckey]) has failed to place \a [held] into this mob's [src.get_index_limb_name(index)]</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to place \a [held] into [src.name]'s ([src.ckey]) [src.get_index_limb_name(index)]</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.name] ([user.ckey]) has failed to place \a [held] into this mob's [src.get_index_limb_name(index)]</font>")
+			log_attack("[user.name] ([user.ckey]) has failed to place \a [held] into [src.name]'s '([src.ckey]) [src.get_index_limb_name(index)]")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			log_attack("[user.name] ([user.ckey]) has failed to place \a [held] into [src.name]'s '([src.ckey]) [src.get_index_limb_name(index)]")
+
+/mob/living/carbon/proc/handle_strip_id(var/mob/living/user)
+	var/obj/item/id_item = src.get_item_by_slot(slot_wear_id)
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+	var/obj/item/place_item = user.get_active_hand()
+	var/pickpocket = user.isGoodPickpocket()
+
+	to_chat(world, "HITLERS: We are [user], stripping [src], our target item is [id_item] in slot [src.slotID2slotname(slot_wear_id)] and our held item is [place_item] please advice")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		if(strip_item_from(user, id_item, slot_wear_id, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has pickpocketed \a [id_item] from [src.name]'s ([src.ckey]) ID slot.</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='red'>Has had \a [id_item] pickpocketed by [user.name] ([user.ckey]) from their ID slot.</font>")
+			log_attack("[user.name] ([user.ckey]) has pickpocketed \a [id_item] from [src.name]'s ([src.ckey]) ID slot.")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		to_chat(user, "<span class='notice'>You try to take [src]'s ID.</span>")
+
+		if(strip_item_from(user, id_item, slot_wear_id, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has pickpocketed \a [id_item] from [src.name]'s ([src.ckey]) ID slot.</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+
+		if(strip_item_from(user, id_item, slot_wear_id, pickpocket))
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has pickpocketed \a [id_item] from [src.name]'s ([src.ckey]) ID slot.</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='red'>Has had \a [id_item] pickpocketed by [user.name] ([user.ckey]) from their ID slot.</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			log_attack("[user.name] ([user.ckey]) has pickpocketed \a [id_item] from [src.name]'s ([src.ckey]) ID slot.")
+			show_inv(user)
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to pickpocket \a [id_item] from [src.name]'s ([src.ckey]) ID slot.</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			show_inv(user)
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to pickpocket \a [id_item] from [src.name]'s ([src.ckey]) ID slot.</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.name] ([user.ckey]) has failed to pickpocket \a [id_item] from this mob's ID slot.</font>")
```
`src.`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+		else
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has failed to pickpocket \a [id_item] from [src.name]'s ([src.ckey]) ID slot.</font>")
+			src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.name] ([user.ckey]) has failed to pickpocket \a [id_item] from this mob's ID slot.</font>")
+			log_attack("[user.name] ([user.ckey]) has failed to pickpocket \a [id_item] from [src.name]'s ([src.ckey]) ID slot.")
```
`src.`

PJB3005 replied:
Okay fuck it let's stop bitching about `src.`

clusterfack commented:
**Suddenly 100+ comments**

Pjb please

PJB3005 commented:
`¯\_(ツ)_/¯`

PJB3005 commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+			log_attack("[user.name] ([user.ckey]) has failed to reverse-pickpocket \a [place_item] into [src.name]'s ([src.ckey]) ID slot.")
+
+/mob/living/carbon/proc/handle_strip_pocket(var/mob/living/user, var/pocket_side)
+	var/pocket_id = (pocket_side == "right" ? slot_r_store : slot_l_store)
```
Make this a define.

xpcybic commented:
oh my fuck what did i just walk into

PJB3005 commented:
`src.`

Jeroen52 commented:
e5a9a4ec4eddf14b7a272f9a75ac4a8d3d19ce2e
Why do you gotta remove the `Hitlers`?

PJB3005 commented on code/modules/mob/mob.dm:
```diff
@@ -494,7 +494,7 @@ var/global/obj/screen/fuckstat/FUCK = new
 			put_in_hand(hand_index, W)
 
-	if(ishuman(src) && W == src:head)
+	if(ishuman(src) && W == src:head) //AAAAAUGH
```
God the least you could do is make this use a cast BECAUSE THIS CODE WILL BREAK IF INVENTORY EVER GETS REFACTORED AND IT WON'T COMPILE ERROR.

9600bauds replied:
Good news, it was completely redundant shitcode anyways!

PJB3005 commented on code/modules/mob/mob.dm:
```diff
@@ -503,90 +503,43 @@ var/global/obj/screen/fuckstat/FUCK = new
+				qdel(W)
+				W = null
+			if(EQUIP_FAILACTION_DROP)
+				W.loc=get_turf(src) // I think.
```
`forceMove()`

9600bauds commented:
//TODO: gets too easily overlooked, instead I use //HITLERS:
Gives unique results and you won't miss a single one because of fear of pushing HITLERS

unid15 commented:
There is literally nothing wrong with using src

9600bauds commented:
I predicted PJB's aneurysm in IRC

PJB3005 commented:
@unid15 yes there is!

9600bauds commented:
Can you explain why?

PJB3005 commented:
Because it does literally nothing.

9600bauds commented:
So there's literally nothing wrong

clusterfack commented:
Src. is for differentiating between locally scoped and src scoped variables and nothing else

9600bauds commented:
I explained above why I used src in some files here, most lines are referencing either the stripper or the strip-ee, with ```user``` being the stripper
Stuff like ```[user.name] ([user.ckey])``` blends in somewhat with ```[name] ([ckey])```, it helps to have the src. there to quickly tell who is this line talking about, especially when you're used to ```user``` being your "main dude".

9600bauds commented:
Basically: Nearly everywhere else you're dealing with item-level or structure-level or datum-level procs, so the "human interaction" bits of code tend to reference ´´´user´´´.
All of this is dealing with user-level procs, which can give you temporary brain farts

PJB3005 commented:
Personally `src.` adds absolutely zero readability. I don't confuse code at all.

clusterfack commented:
Coder fight

Probe1 commented:
 >100+ comments in 2 hours
I am terrified.

DrCelt commented:
src. when not needed is the same as a trailing return, remove. remove remove

9600bauds commented:
Parkinson's law of triviality is C. Northcote Parkinson's 1957 argument that members of an organisation give disproportionate weight to trivial issues.He observed that a committee whose job was to approve the plans for a nuclear power plant spent the majority of its time on discussions about relatively minor but easy-to-grasp issues, such as what materials to use for the staff bike-shed, while neglecting the proposed design of the plant itself, which is far more important but also a far more difficult and complex task.

The-Albinobigfoot commented:
roasted

dylanstrategie commented:
>"//Our inventory procs should be able to handle the following, but our inventory code is hot spaghetti bologni, so here we go"

Oh hey, I wrote that. I think it was with the explosive handcuffs thing ? Did it ever happen ? Guess the actual code handler happened in the meantime

PJB3005 commented:
@dylanstrategie yeah I revived them.

TheFriggenSawbones commented:
why cant you guys put this kinda work into space fishing

ComicIronic commented on code/modules/mob/living/carbon/martian/martian.dm:
```diff
@@ -94,9 +94,9 @@
-				W.dropped(src)
+			W.forceMove(loc)
+			W.unequipped()
+			if(dropped) W.dropped(src)
```
No single line ifs

ComicIronic commented on code/modules/mob/living/carbon/stripping.dm:
```diff
@@ -0,0 +1,267 @@
+	var/obj/item/place_item = user.get_active_hand()
+	var/pickpocket = user.isGoodPickpocket()
+
+	to_chat(world, "HITLERS: We are [user], stripping [src], our target item is [pocket_item] in slot [src.slotID2slotname(pocket_id)] [pocket_side] pocket .and our held item is [place_item] please advice")
```
That's a nice hitlers

Kurfursten commented:
At first I thought this was a highly controversial PR because it got 150 posts in 8 hours

Then I saw it's all just PJB saying "src" over and over again :eggplant:

9600bauds commented:
Well this is honestly pretty goddamn annoying, line notes on this PR are horrible to check because there's a huge monolith of "```src```" that splits the sea like Moses so people don't even see the replies way up there.
This is worse than a waste of time. The actual code review is being inconvenienced by this. Anything that could warrant attention is drowned out by this src nonsense.

Exxion commented:
Since when is `src.` considered a problem, anyway? It neither helps nor hurts readability and it has exactly zero effect on the code.

9600bauds commented:
Sorry if this offends anyone but I deleted all that noise, if I want to touch up the srcs I know where to find them anyways

DrCelt commented:
>deleting comments which say your PR is bad

wow how do I get collaborator to make it seem like hypothermia is well recieved

Shadowmech88 commented:
The extra instances of `src` *do* slightly increase the file size.

9600bauds commented:
There's plenty of comments pointing out ´´´src.´´´, we absolutely do not need SIXTY comments saying the same thing taking up two thirds of the page

Jeroen52 commented:
@9600bauds I believe that they were pointing out individual cases or `src`.

9600bauds commented:
I know where to find them, I used ```src.``` in two files on purpose. Not even that anyways because PJB mercifully stopped halfway through.

clusterfack commented:
This PR has become a meme pr

9600bauds commented:
And to think we make fun of TG for typeless loops.

Exxion commented:
@Shadowmech88 If there are five instances of `src.` in *every single file of the entire repo*
Then they take up just over .01% of the filesize

Shadowmech88 commented:
Hey, that .01% adds up.

DrCelt commented:
Imagine if we had 100 times more code - then it'd be 1% of the filesize. And, god forbid, imagine we had 100 times that. It'd become 100% of the filesize. How would we code then, @Exxion ? There'd be no room. Checkmate.

TheFriggenSawbones commented:
lets do some math here,

(0.01*1.8)+32=32.018?

Hope that helps the europeans

DrCelt commented:
Oh, I forgot he was using fahrenheit percentages. Man, 32% of the repo is pretty massive, gotta be honest here.

Exxion commented:
fukc... i cant argue with that logic

invoid9 commented:
Shit, we're gonna have to remove ALL examples of Fahrenheit percentages.

I'll start pointing them out in this PR, hold on, lemme copy paste Fahrenheit percentages, this will take a bit.

unid15 commented:
Remove comments when

They add zero readability (unless you confuse code like some kind of a pleb), do literally nothing and increase file size

:^)

Aurantiaco commented:
> Buffed the Pickpocket's Gloves: Their effects now apply to all storage slots, not just ID and pockets.
Oh hello officer that's a nice security belt and taser in your suit storage. YOINK

Shadowmech88 commented:
Well at least with guns and such you'd be able to see it appear on their sprite once it's put in their hand.

Aurantiaco commented:
Also with guns the person will probably be tased/shot before they can react

9600bauds commented:
You could convince the HoS to let you stand right next to him for 3 seconds and hope he doesn't notice you stole his shit
or
OR
you could stand right next to him for 0.2 seconds and chloral him or parasting him or memepen him or glare him
which is why they won't let you stand next to them

The pickpocket's gloves are criminally underused as they are right now

Aurantiaco commented:
You sit next to him at the bar and he doesn't think anything of it.

Then you steal everything and kill him dead

9600bauds commented:
That scenario is so less likely and less practical and more preventable and harder to pull off and more visible than so many other things you could do to kill the HoS, some of them right there in your uplink, the hell

Aurantiaco commented:
Parapen is for unrobust faggots. If you buy parapens then you should quit immediately.

9600bauds commented:
Agreed, that's why having more semi-viable options is good

Shadowmech88 commented:
Really, if you manage to get next to the HoS without suspicion, even slipping him with syndie soap and cablecuffing him would be almost as effective as stripping his gun off him, stunning him, and then cablecuffing him.

dylanstrategie commented:
Buffing the Pickpocket Gloves is fine

If you're stupid enough to stand still near an Assistant in a public area, you deserve to be robbed blind

Plus, it's one item at a time. Either he steals your gun and stuns you, or steals your ID and flees. And if he's caught, pickpocketing won't save him

DrCelt commented on code/game/objects/items/weapons/dna_injector.dm:
```diff
@@ -127,90 +126,38 @@
-					else
-						user.show_message(text("<span class='warning'>You finish the injection.</span>"))
-	return
+			if (GetState(MONKEYBLOCK) && istype(M, /mob/living/carbon/human))
```
atomic

9600bauds replied:
literally the old logging verbatim as it was

DrCelt commented on code/modules/mob/living/carbon/internals.dm:
```diff
@@ -0,0 +1,68 @@
+		if(internals) //This is the HUD icon, these variables have WAY too similar names
+			internals.icon_state = "internal0"
+		if(user != src)
+			if(!user.isGoodPickpocket())
```
Jesus christ, that's awful.

ComicIronic replied:
What's wrong with it

9600bauds replied:
What's wrong with it

ComicIronic replied:
What's wrong with it

9600bauds replied:
Where's the bomb

optimumtact replied:
rename the proc to isBadPickPocket and flip the return value

boom.

9600bauds replied:
Is that it? I can't tell what he's talking about

optimumtact replied:
no I have no idea honestly

DrCelt replied:
I wasn't insulting the code I just thought this is evil. Taking people's internals?

9600bauds replied:
How are you gonna sneakily gas the Vox if they notice you doing it?

Exxion commented:
Hold on, I recognize that first concerning comment
Isn't that from MEGAMADCRAB code?

9600bauds commented:
Promise I'll get off my ass and finish this soon

Probe1 commented:
Put the spurs to it and don't fail.  This is good work and needs done.

9600bauds commented:
Been putting off my edjucation for too long now but I'll pick this back up tomorrow

DrCelt commented:
you're holding back PLANES @9600bauds 
:^)

9600bauds commented:
Call it jetlag

9600bauds commented:
Well I tested and fixed up everything I had in mind with this and removed Hitler again, going to sleep for today. Only thing I couldn't figure out was making the stripping window not re-open itself when refreshing after a successful action, but that's fairly minor.
If nobody has anything to say, I'll look at this again tomorrow, then I think I'm done

9600bauds commented:
\>conflicts

![child-inhales-dx3dby](https://cloud.githubusercontent.com/assets/6526157/16106667/c4452d96-336c-11e6-8d07-06d23d856230.jpg)

tomorrow

9600bauds commented on code/modules/mob/inventory.dm:
```diff
@@ -158,15 +158,12 @@
 	if(!is_valid_hand_index(index) || !is_valid_hand_index(active_hand))
 		return 0
 
-	if(held_items[index])
```
moved directly into put_in_hand_check()

9600bauds commented:
Seems good, tested a bunch

Probe1 commented:
we did it reddit!

9600bauds commented:
who's ready for all the new bugs this will inevitably cause?!

Shadowmech88 commented:
Bugs are the spice of spess.

Probe1 commented:
![9567347](https://cloud.githubusercontent.com/assets/8550416/16168353/443275fc-34d2-11e6-86a9-79456801047c.jpg)
